Guides/Grouped.md

```diff
@@ -0,0 +1,71 @@
+# AdjacentPairs
+
+[[Source](https://github.com/apple/swift-algorithms/blob/main/Sources/Algorithms/Grouped.swift) | 
+ [Tests](https://github.com/apple/swift-algorithms/blob/main/Tests/SwiftAlgorithmsTests/GroupedTests.swift)]
+ 
+Groups up elements of a sequence into a new Dictionary, whose values are Arrays of grouped elements, each keyed by the result of the given closure.
+
+This operation is available for any sequence by calling the `grouped(by:)`
+method.
+
+```swift
+let fruits = ["Apricot", "Banana", "Apple", "Cherry", "Avocado", "Coconut"]
+let fruitsByLetter = fruits.grouped(by: { $0.first! })
+// Results in:
+[
+    "B": ["Banana"],
+    "A": ["Apricot", "Apple", "Avocado"],
+    "C": ["Cherry", "Coconut"],
+]
```

Comment:
This is a different style to how other examples show their output, but I think it's better to be multi-line and syntax highlighted in this case. Compare:

```swift
// fruitsByLetter == ["B": ["Banana"],  "A": ["Apricot", "Apple", "Avocado"], "C": ["Cherry", "Coconut"]]
```